src/st/st-theme-node-private.h

```diff
@@ -44,6 +44,12 @@ struct _StThemeNode {
   StGradientType background_gradient_type;
   ClutterColor background_gradient_end;
 
+  guint background_gradient_position_set : 1;
+  int background_gradient_position_x;
+  int background_gradient_position_y;
+  int background_gradient_stop;
```

Comment:
I would call this `_color_stop`, since that's what it is, and how both Cairo and CSS calls those types of stops. Also, as mentioned before, it would be nice if this could support more than one stop, since that might make the patch more upstream friendly (although I'm happy to leave that for a separate PR, if that works better)

Reply:
Hmm... actually my comment is not correct: this represents the position relative (between 0.0 and 1.0) to the width of the element it's being applied on top, so color-stop might not be a good name. Maybe stop-position?

Or even better, make it  as in [CSS's radial-gradient](https://developer.mozilla.org/en-US/docs/Web/CSS/radial-gradient) and accept the color + the position of the stop, instead of using `node->background_color`.